docs/database.rst

```diff
@@ -0,0 +1,40 @@
+rotki Database
+##############
```

Comment:
Maybe have it as documentation for developers explaining parts of the design and having this as subsection? What do you think @LefterisJP

Reply:
Done :heavy_check_mark: